src/Time/TimeSteppers/ImexTimeStepper.hpp

```diff
@@ -0,0 +1,177 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include "DataStructures/DataBox/PrefixHelpers.hpp"
+#include "DataStructures/DataBox/Prefixes.hpp"
+#include "DataStructures/MathWrapper.hpp"
+#include "Time/History.hpp"
+#include "Time/Time.hpp"
+#include "Time/TimeSteppers/TimeStepper.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/Serialization/CharmPupable.hpp"
+
+/// \cond
+#define IMEX_TIME_STEPPER_WRAPPED_TYPE(data) BOOST_PP_TUPLE_ELEM(0, data)
+#define IMEX_TIME_STEPPER_DERIVED_CLASS(data) BOOST_PP_TUPLE_ELEM(1, data)
+/// \endcond
+
+/*!
+ * \ingroup TimeSteppersGroup
+ *
+ * Base class for TimeSteppers with IMEX support, derived from
+ * TimeStepper.
+ *
+ * All supported time-stepping algorithms (both implicit and
+ * explicit) consist of substeps that add linear combinations of
+ * derivative values.  For implicit substeps, it is convenient to
+ * split out the contribution of the final implicit term from
+ * contributions from the history.  We therefore write an implicit
+ * substep as
+ *
+ * \f{equation}{
+ *   Y_{n+1} =
+ *   Y_{n,\text{explicit}} + Y_{n,\text{inhomogeneous}} + w_n S(Y_{n+1})
+ * \f}
+ *
+ * Here \f$Y_{n,\text{explicit}}\f$ is the value of the evolved
+ * variables before the implicit substep is applied,
+ * \f$Y_{n,\text{inhomogeneous}}\f$ is the contribution of the past
+ * values from the implicit derivative history, and \f$S(\cdot)\f$ is
+ * the implicit portion of the right-hand-side (generally source
+ * terms for PDEs).  We call \f$w_n\f$ the \emph{implicit weight}.
```

Comment:
Doxygen doesn't like `\emph`.  Need to replace it with something else.  (Not sure how I messed that locally).